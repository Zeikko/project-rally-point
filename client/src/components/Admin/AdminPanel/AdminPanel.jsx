import _ from 'lodash'
import PropTypes from 'prop-types'
import React from 'react'
import * as propTypes from '../../../constants/prop-types'
import SimulateQueueFull from '../SimulateQueueFull/SimulateQueueFull'
import SimulateQueueEmpty from '../SimulateQueueEmpty/SimulateQueueEmpty'

function AdminPanel(props) {
  const {
    dispatch, user, game,
  } = props
  if (_.get(user, 'data.role') !== 'admin') {
    return null
  }
  return (
    <div>
      <SimulateQueueFull game={game} dispatch={dispatch} />
      <SimulateQueueEmpty game={game} dispatch={dispatch} />
    </div>
  )
}

AdminPanel.propTypes = {
  dispatch: PropTypes.func.isRequired,
  user: propTypes.user.isRequired,
  game: propTypes.game.isRequired,
}

export default AdminPanel
