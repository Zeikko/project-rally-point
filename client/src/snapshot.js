import renderer from 'react-test-renderer'
import 'jest-styled-components'

export function matchComponentToSnapshot(component) {
  const json = renderer.create(component).toJSON()
  expect(json).toMatchSnapshot()
}