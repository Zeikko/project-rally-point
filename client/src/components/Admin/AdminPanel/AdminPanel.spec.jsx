import React from 'react'
import AdminPanel from './AdminPanel'
import { matchComponentToSnapshot } from '../../../test/snapshot'
import initialState from '../../../initial-state'
import adminUser from '../../../../../fixtures/admin-user.json'
import normalUser from '../../../../../fixtures/normal-user.json'

describe('AdminPanel', () => {
  it('renders initial state correctly', () => {
    matchComponentToSnapshot(<AdminPanel
      dispatch={() => {}}
      {...initialState}
      user={{ isLoading: false, data: normalUser }}
    />)
  })

  it('renders correctly admin user', () => {
    matchComponentToSnapshot(<AdminPanel
      dispatch={() => {}}
      {...initialState}
      user={{ isLoading: false, data: adminUser }}
    />)
  })
})
