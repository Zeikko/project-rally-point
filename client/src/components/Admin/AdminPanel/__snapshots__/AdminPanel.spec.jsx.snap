// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanel renders correctly admin user 1`] = `
.c0 {
  font-size: 18px;
  border: solid black 1px;
  background-color: #AAA;
  cursor: pointer;
}

<div>
  <button
    className="c0"
    onClick={[Function]}
  >
    Simulate Queue Full
  </button>
  <button
    className="c0"
    onClick={[Function]}
  >
    Simulate Queue Empty
  </button>
</div>
`;

exports[`AdminPanel renders initial state correctly 1`] = `null`;
